.github/workflows/build.yaml

```diff
@@ -15,23 +15,86 @@ permissions: read-all
 
 env:
   AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  ACTIONS_RUNNER_DEBUG: true
 jobs:
+  flutter-prep:
+    name: ${{ matrix.os }} Flutter Prep
+    outputs:
+      latest_flutter_candidate: ${{ steps.flutter-candidate.outputs.FLUTTER_CANDIDATE }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: git clone devtools
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+
+      - name: Get Latest Flutter Candidate
+        id: flutter-candidate
+        run: |
+          LATEST_FLUTTER_CANDIDATE=$(./tool/latest_flutter_candidate.sh)
+          echo "FLUTTER_CANDIDATE=$LATEST_FLUTTER_CANDIDATE" >> $GITHUB_OUTPUT
+      
+      - name: Load Cached Flutter SDK
+        id: cache-flutter
+        uses: actions/cache@v3
+        with:
+          path: |
+            ./flutter-sdk
+          key: flutter-sdk-${{ runner.os }}-${{ steps.flutter-candidate.outputs.FLUTTER_CANDIDATE }}
+
+      - if: ${{ steps.cache-flutter.outputs.cache-hit != 'true' }}
+        name: Clone Flutter SDK if none cached
+        run: |
+          git clone https://github.com/flutter/flutter.git ./flutter-sdk
+          cd flutter-sdk
+          git checkout $LATEST_FLUTTER_CANDIDATE
+        env:
+          LATEST_FLUTTER_CANDIDATE: ${{ steps.flutter-candidate.outputs.FLUTTER_CANDIDATE }}
```

Comment:
I see ${{ steps.flutter-candidate.outputs.FLUTTER_CANDIDATE }} being used a few times, are we duplicating this, or in other words can it be put in a local var?

Reply:
unfortunately I don't think there is a way to make this less verbose.